modules/charon-core/src/test/java/org/wso2/charon3/core/protocol/endpoints/MeResourceManagerTest.java

```diff
@@ -0,0 +1,1547 @@
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(CharonException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new CharonException()));
+        when(userManager.getMe(name, requiredAttributes)).thenThrow(CharonException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForGetBadRequestException")
+    public Object[][] dataToGetCharonBadRequestException() {
+
+        return new Object[][]{
+                {"David", "userName", null, ResponseCodeConstants.CODE_BAD_REQUEST}
+        };
+    }
+
+    @Test(dataProvider = "dataForGetBadRequestException")
+    public void testGetUserBadRequestException(String name, String attributes, String excludeAttributes,
+                                               int expectedScimResponseStatus)
+            throws CharonException, BadRequestException, NotFoundException {
+
+        SCIMResourceTypeSchema schema = SCIMResourceSchemaManager.getInstance().getUserResourceSchema();
+        Map<String, Boolean> requiredAttributes = ResourceManagerUtil.getOnlyRequiredAttributesURIs(
+                (SCIMResourceTypeSchema)
+                        CopyUtil.deepCopy(schema), attributes, excludeAttributes);
+
+        mockStatic(AbstractResourceManager.class);
+
+        when(AbstractResourceManager.getResourceEndpointURL(SCIMConstants.USER_ENDPOINT))
+                .thenReturn(ENDPOINT_ME);
+        when(AbstractResourceManager.getEncoder()).thenReturn(new JSONEncoder());
+        when(AbstractResourceManager.getDecoder()).thenReturn(new JSONDecoder());
+        when(AbstractResourceManager.encodeSCIMException(any(BadRequestException.class)))
+                .thenReturn(getEncodeSCIMExceptionObject(new BadRequestException()));
+        when(userManager.getMe(name, requiredAttributes)).thenThrow(BadRequestException.class);
+
+        SCIMResponse outputScimResponse = meResourceManager.get(name, userManager, attributes, excludeAttributes);
+        Assert.assertEquals(outputScimResponse.getResponseStatus(), expectedScimResponseStatus);
+
+    }
+
+    @DataProvider(name = "dataForTestCreateSuccess")
+    public Object[][] dataToTestCreateSuccess() throws BadRequestException, CharonException, InternalErrorException {
+
+        User user = getNewUser();
+
+        return new Object[][]{
+                {NEW_USER_SCIM_OBJECT_STRING, "userName", null, user, ResponseCodeConstants.CODE_CREATED}
+        };
+    }
+
+    @Test(dataProvider = "dataForTestCreateSuccess")
+    public void testCreateSuccess(String scimObjectString, String attributes, String excludeAttributes,
```

Comment:
testCreateUserSuccess

Reply:
Fixed in commit a5699e694bc31369fb4a347d4a0a12177e73775d